packages/react-core/src/components/ToggleGroup/ToggleGroup.tsx

```diff
@@ -17,12 +20,15 @@ export const ToggleGroup: React.FunctionComponent<ToggleGroupProps> = ({
   className,
   children,
   isCompact = false,
+  disableAll = false,
   'aria-label': ariaLabel,
   ...props
 }: ToggleGroupProps) => {
-  const toggleGroupItemList = [] as any[];
-  React.Children.forEach(children, child => {
-    toggleGroupItemList.push(child);
+  const toggleGroupItemList = React.Children.map(children, child => {
+    const childCompName = (child as any).type.name;
+    return childCompName !== ToggleGroupItem.name
+      ? child
+      : React.cloneElement(child as React.ReactElement, disableAll ? { isDisabled: true } : {});
```

Comment:
We try and avoid cloneElement when we can.  Can we do this with a render prop instead?

Reply:
In the current state, I don't think so. The closest we can do is to create ToggleGroupContext which will pass this to other ToggleGroupItems through its context.
What do you think?

Reply:
Hmmm.   That might be a better option @boaz0 consumers have asked we avoid cloneElement as it can be problematic.

Reply:
So you want me to do go with the context idea and revert the current changes I've made?

Reply:
I think either method works in this case because we're type checking the child before cloning

Reply:
@boaz0 I think your approach is fine for this use case.